util/types/time.go

```diff
@@ -871,7 +879,10 @@ func ParseDuration(str string, fsp int) (Duration, error) {
 			hour, err = strconv.Atoi(seps[0])
 		} else {
 			// No delimiter.
-			if len(str) == 6 {
+			if len(str) == 7 {
```

Comment:
I didn't change this, I add a check for Duration like `HHHMMSS`, because the Time type in MySQL ranges from -838:59:59.000000 to 838:59:59.000000.